python/ray/data/preprocessors/imputer.py

```diff
@@ -131,7 +131,14 @@ def _transform_pandas(self, df: pd.DataFrame):
                 if is_categorical_dtype(df.dtypes[column]):
                     df[column] = df[column].cat.add_categories(value)
 
-        df = df.fillna(new_values)
+        for column_name in new_values:
```

Comment:
can you add this behavior to the docstring as well?

Reply:
done!